### PR TITLE
feat(anta.tests): Adding the test case to verify BFD session and timers

### LINKED ISSUE #499: Add the test case to verify the BFD session, timers and peers

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
# Description

Add the test case to verify the BFD session and timers.

Run the testcase by adding catalog as below:
```
anta.tests.bfd:
  - VerifyBFDSpecificPeers:
      bfd_peers:
        - peer_address: 192.0.255.8
          vrf: default
        - peer_address: 192.0.255.7
          vrf: default
  - VerifyBFDPeersIntervals:
      bfd_peers:
        - peer_address: 192.0.255.8
          vrf: default
          tx_interval: 1200000
          rx_interval: 1200000
          multiplier: 3
        - peer_address: 192.0.255.7
          vrf: default
          tx_interval: 1200000
          rx_interval: 1200000
          multiplier: 3
```

1. Testcase pass when details are correct.
```
                                                                    All tests results                                                                     
┏━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┓
┃ Device    ┃ Test Name               ┃ Test Status ┃ Message(s) ┃ Test description                                                       ┃ Test category ┃
┡━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━┩
│ s1-spine1 │ VerifyBFDPeersIntervals │ success     │            │ Verifies the timers of the BFD peers in the specified VRF.             │ bfd           │
├───────────┼─────────────────────────┼─────────────┼────────────┼────────────────────────────────────────────────────────────────────────┼───────────────┤
│ s1-spine1 │ VerifyBFDSpecificPeers  │ success     │            │ Verifies the BFD peer's sessions and remote disc in the specified VRF. │ bfd           │
└───────────┴─────────────────────────┴─────────────┴────────────┴────────────────────────────────────────────────────────────────────────┴───────────────┘
```

2. Testcase fails when bfd neighbor  not found.
```

                                                                                          All tests results                                                                                           
┏━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┓
┃ Device    ┃ Test Name               ┃ Test Status ┃ Message(s)                                                     ┃ Test description                                              ┃ Test category ┃
┡━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━┩
│ s1-spine1 │ VerifyBFDSpecificPeers  │ failure     │ Following BFD peers are not configured, status is not up or    │ Verifies the BFD peer's sessions and remote disc in the       │ bfd           │
│           │                         │             │ remote disc is zero:                                           │ specified VRF.                                                │               │
│           │                         │             │ {'192.0.255.81': {'MGMT': 'Not Configured'}, '192.0.255.71':   │                                                               │               │
│           │                         │             │ {'default': 'Not Configured'}}                                 │                                                               │               │
├───────────┼─────────────────────────┼─────────────┼────────────────────────────────────────────────────────────────┼───────────────────────────────────────────────────────────────┼───────────────┤
│ s1-spine1 │ VerifyBFDPeersIntervals │ failure     │ Following BFD peers are not configured or timers are not       │ Verifies the timers of the BFD peers in the specified VRF.    │ bfd           │
│           │                         │             │ correct:                                                       │                                                               │               │
│           │                         │             │ {'192.0.255.81': {'MGMT': 'Not Configured'}, '192.0.255.71':   │                                                               │               │
│           │                         │             │ {'default': 'Not Configured'}}                                 │                                                               │               │
└───────────┴─────────────────────────┴─────────────┴────────────────────────────────────────────────────────────────┴───────────────────────────────────────────────────────────────┴───────────────┘
```

3. Testcase fails when BDF neighbor session is not up and timers are not correct.
```
 
                                                                                          All tests results                                                                                           
┏━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┓
┃ Device    ┃ Test Name               ┃ Test Status ┃ Message(s)                                                     ┃ Test description                                              ┃ Test category ┃
┡━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━┩
│ s1-spine1 │ VerifyBFDPeersIntervals │ failure     │ Following BFD peers are not configured or timers are not       │ Verifies the timers of the BFD peers in the specified VRF.    │ bfd           │
│           │                         │             │ correct:                                                       │                                                               │               │
│           │                         │             │ {'192.0.255.8': {'default': {'tx_interval': 1200000,           │                                                               │               │
│           │                         │             │ 'rx_interval': 1200000, 'multiplier': 3}}, '192.0.255.7':      │                                                               │               │
│           │                         │             │ {'default': {'tx_interval': 1200000, 'rx_interval': 1200000,   │                                                               │               │
│           │                         │             │ 'multiplier': 3}}}                                             │                                                               │               │
├───────────┼─────────────────────────┼─────────────┼────────────────────────────────────────────────────────────────┼───────────────────────────────────────────────────────────────┼───────────────┤
│ s1-spine1 │ VerifyBFDSpecificPeers  │ failure     │ Following BFD peers are not configured, status is not up or    │ Verifies the BFD peer's sessions and remote disc in the       │ bfd           │
│           │                         │             │ remote disc is zero:                                           │ specified VRF.                                                │               │
│           │                         │             │ {'192.0.255.8': {'default': {'status': 'down', 'remote_disc':  │                                                               │               │
│           │                         │             │ 0}}, '192.0.255.7': {'default': {'status': 'down',             │                                                               │               │
│           │                         │             │ 'remote_disc': 0}}}                                            │                                                               │               │
└───────────┴─────────────────────────┴─────────────┴────────────────────────────────────────────────────────────────┴───────────────────────────────────────────────────────────────┴───────────────┘
```

Fixes #499 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/bfd.py
Comment (carl-baillargeon) [inline anta/tests/bfd.py:1]: I know it's not part of this PR but we should bring and maybe refactor the current BFD test:
https://www.anta.ninja/stable/api/tests.routing.generic/#anta.tests.routing.generic.VerifyBFD

File: anta/tests/bfd.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        class BFDNeighbors(BaseModel):
+            """
+            This class defines the details of a BFD neighbor.
+            """
+
+            neighbor: Union[IPv4Address, IPv6Address]
+            """IPv4/IPv6 BFD neighbor"""
+            vrf: str = "default"
+            """VRF context"""
+            loopback: Union[IPv4Address, IPv6Address]
```
Comment (carl-baillargeon) [inline anta/tests/bfd.py:52]: Is it appropriate to call this `loopback` or can it be any IP?

**Review by carl-baillargeon — CHANGES_REQUESTED** [review]

File: anta/tests/bfd.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            peer_address: IPv4Address
+            """IPv4 address of a BFD peer"""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+            tx_interval: int
+            """Tx interval of BFD peer"""
+            rx_interval: int
+            """Rx interval of BFD peer"""
+            multiplier: int
+            """Multiplier of BFD peer"""
```
Comment (carl-baillargeon) [inline anta/tests/bfd.py:113]: Can we be more restrictive for these inputs?

**Review by carl-baillargeon — APPROVED** [review]
LGTM. Please check my comment on inputs if it's possible or not.

File: anta/tests/bfd.py (hunk 2 above)
Comment (MaheshGSLAB) [reply anta/tests/bfd.py]: Yes we can, added intervals and multiplier in custom output

**Review by titom73 — APPROVED** [review]
LGTM

**Pull request merged**